sdk/lib/rtl/heapuser.c

```diff
@@ -127,3 +127,32 @@ RtlInitializeHeapManager(VOID)
     RtlInitializeCriticalSection(&RtlpProcessHeapsListLock);
 }
 
+ULONG NTAPI
+RtlGetProcessHeaps(_In_ ULONG HeapCount, _Out_cap_(HeapCount) HANDLE *HeapArray)
+{
+    PPEB Peb = RtlGetCurrentPeb();
+
+    RtlEnterCriticalSection(&RtlpProcessHeapsListLock);
+    ULONG nHeaps = Peb->NumberOfHeaps;
+
+    _SEH2_TRY
+    {
+        if (HeapArray)
```

Comment:
Would it be worth to adapt this code so `_SEH2_TRY ...` is inside this `if()`?

Reply:
I see no reason why